### PR TITLE
Fix swagger basePath

### PR DESCRIPTION
**Pull request by Ener-Getick**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/api-platform/core/issues/722
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Swagger/Serializer/DocumentationNormalizer.php [hunk 1]
```diff
@@ -41,15 +42,17 @@
     private $resourceClassResolver;
     private $operationMethodResolver;
     private $operationPathResolver;
+    private $requestContext;
 
-    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceClassResolverInterface $resourceClassResolver, OperationMethodResolverInterface $operationMethodResolver, OperationPathResolverInterface $operationPathResolver)
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceClassResolverInterface $resourceClassResolver, OperationMethodResolverInterface $operationMethodResolver, OperationPathResolverInterface $operationPathResolver, RequestContext $requestContext)
```
Comment (Ener-Getick) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: if you prefer, we can provide easily a bc layer by instantiating the request context with its default values.

Comment (dunglas) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: We don't care about BC for now, so it's ok :)

Comment (dunglas) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: However adding a dependency to the Symfony Router in this component is not ok.

Comment (Ener-Getick) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: arf indeed... we could have something like a proxy to the `RequestContext`?

Comment (Ener-Getick) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: Or we can do this in the `RouterOperationPathResolver` but then resolvers result won't be consistent anymore.

Comment (Ener-Getick) [inline src/Swagger/Serializer/DocumentationNormalizer.php]: Or maybe better:
```php
if ($this->operationPathResolver instanceof RouterOperationPathResolver) {
    $doc['basePath'] = $this->operationPathResolver->getContext()->getBaseUrl();
}
```
What do you prefer?

**Comment by dunglas** [comment]
An alternative (maybe cleaner) would be to allow to inject the `basePath` as a `string` in the constructor (can be useful in component context too) and introduce a new BasePathFactory in the Symfony Routing Bridge to return the base path using the router's context.
Then use this factory in the DI to inject the base path when using Symfony.

WDYT @Ener-Getick?

**Comment by dunglas** [comment]
Thanks for working on this @Ener-Getick!

**Comment by Ener-Getick** [comment]
You're welcome :)

**Pull request closed without merging**